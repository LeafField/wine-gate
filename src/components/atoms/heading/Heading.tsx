import React, { FC } from "react";

type Props = {
  title: string;
};

const Heading: FC<Props> = ({ title }) => {
  return <h2 className="text-heading2 border-b-2 border-red  ">{title}</h2>;
};

export default Heading;
