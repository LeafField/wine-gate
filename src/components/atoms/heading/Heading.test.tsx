import { render, screen } from "@testing-library/react";
import Heading from "./Heading";
import { describe, test, expect } from "vitest";

describe("Headingの単体テスト", () => {
  test("タイトルが正しく表示される", () => {
    const testTitle = "テストタイトル";
    render(<Heading title={testTitle} />);
    const headingElement = screen.getByText(testTitle);
    expect(headingElement).toBeDefined();
  });

  test("スナップショットテスト", () => {
    const testTitle = "テストタイトル";
    const { container } = render(<Heading title={testTitle} />);
    expect(container).toMatchSnapshot();
  });
});
