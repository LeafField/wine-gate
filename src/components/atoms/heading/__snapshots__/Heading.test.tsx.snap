// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headingの単体テスト スナップショットテスト 1`] = `
<div>
  <h2
    class="text-heading2 border-b-2 border-red  "
  >
    テストタイトル
  </h2>
</div>
`;
